runbooks/kafka-rebalance.md

```diff
@@ -0,0 +1,168 @@
+- Existing brokers continue to handle the full load.
+- Rebalancing spreads the load and improves performance and fault tolerance.
+
+---
+
+## 🔧 Step-by-Step: Kafka Rebalancing Guide
+
+### ✅ Step 0: Get the Broker List from ZooKeeper
+
+Use it if partitions are imbalanced. You can check the data directory to see if there is imbalance.
+
+Find the # of replicas and their usage
+
+```bash
+$ num_replicas=$(kubectl get statefulsets.apps kafka | grep -v "NAME" | awk '{print $2}' | awk -F'/' '{print $2}')
+
+$ for i in 0 1 2 3 4 5 6 7 8;do echo "usage for kafka-broker-${i}"; kubectl exec -it kafka-broker-${i} -- du -hd1 /bitnami/kafka/data | grep -v "__consumer" | grep -v "__transaction" ; done
+```
+
+
+### ✅ Step 1: Get the Broker List from ZooKeeper
+
+If you're using ZooKeeper (not KRaft mode), get the list of broker IDs like this:
+
+```bash
+zkCli.sh -server localhost:2181
+```
+
+Then run inside the shell:
+
+```bash
+ls /brokers/ids
+```
+
+You'll get output like:
+
+```
+[100, 101, 102, 103]
+```
+
+Use these broker IDs in your reassignment plan.
+
+Exit the shell with:
+
+```bash
+quit
+```
+
+---
+
+### ✅ Step 2: View Current Partition Distribution
+
+You can inspect how partitions are currently distributed across brokers:
+
+```bash
+kubectl exec -ti -n kfuse kafka-broker-0 -- bash
+unset JMX_PORT
+/opt/bitnami/kafka/bin/kafka-topics.sh --bootstrap-server :9092 --list
+```
+
+---
+
+### ✅ Step 3: Create a Topics JSON File
+
+You’ll need a JSON file listing the topics to rebalance. Create a file called `topics.json`:
+
+```json
+{
+  "version": 1,
+  "topics": [
+    { "topic": "kf_events_topic" },
+    { "topic": "kf_logs_metric_topic" },
+    { "topic": "kf_logs_topic" },
+    { "topic": "kf_metrics_topic" },
+    { "topic": "kf_traces_errors_topic" },
+    { "topic": "kf_traces_metric_topic" },
+    { "topic": "kf_traces_topic" },
+    { "topic": "logs_ingest_topic" }
+  ]
+}
+```
+---
+
+### ✅ Step 4: Generate a Reassignment Plan
+
+Now generate a plan using the list of broker IDs you got earlier:
+
+```bash
+kafka-reassign-partitions.sh \
+  --bootstrap-server <broker-host>:9092 \
+  --generate \
+  --topics-to-move-json-file topics.json \
+  --broker-list "100,101,102,103" > reassign-plan.json
+```
+
+This writes a JSON plan to `reassign-plan.json`.
+
+---
+
+### ✅ Step 5: Review the Plan
+
+Open and review `reassign-plan.json`. It should look like:
+
+```json
+{
+  "version": 1,
+  "partitions": [
+    {
+      "topic": "your-topic",
+      "partition": 0,
+      "replicas": [2, 3],
+      "log_dirs": ["any", "any"]
+    }
+  ]
+}
+```
+
+The above command will print out the `Current partition replica assignment` and `Proposed partition reassignment configuration`. Ignore the proposed output. Only the Current partition replica assignment is needed. Format the `Current partition replica assignment` in standard JSON format and make sure it looks reasonable and distributes partitions across all brokers.
```

Comment:
@swapnil-kfuse We want the proposed one right, and no the current replica assignment?